app/services/sources/properties-managers/properties-manager.ts

```diff
@@ -0,0 +1,137 @@
+import * as input from 'components/shared/forms/Input';
+import * as obs from '../../../../obs-api';
+
+
+/**
+ * This is the interface that the rest of the app uses
+ * to interact with property managers in a generic way.
+ */
+export interface IPropertyManager {
+  destroy(): void;
+  getPropertiesFormData(): input.TFormData;
+  setPropertyFormData(property: TCustomProperty |
+    input.IFormInput<input.TObsValue> |
+    input.IListInput<input.TObsValue>): void;
+  settings: Dictionary<any>;
+}
+
+
+export interface ICustomProperty<TValue> extends input.IFormInput<TValue> {
+  isCustom: true;
+}
```

Comment:
Are you sure that here can be something different than `TObsValue` ?

Reply:
👍 I restricted `TValue` so it must be a `TObsValue`